smartsim/wlm/slurm.py

```diff
@@ -273,6 +274,29 @@ def _get_alloc_cmd(
     return salloc_args
 
 
+def _validate_time_format(time: str) -> str:
+    """Convert time into valid walltime format
+
+    By defualt the formatted wall time is the total number of seconds.
+
+    :param time: number of hours to run job
+    :type time: str
+    :returns: Formatted walltime
+    :rtype: str
+    """
+    try:
+        hours, minutes, seconds = map(int, time.split(":"))
+    except ValueError as e:
+        raise ValueError(
+            "Input time must be formatted as `HH:MM:SS` with valid Integers."
+        ) from e
+    delta = datetime.timedelta(hours=hours, minutes=minutes, seconds=seconds)
+    fmt_str = str(delta)
+    if delta.seconds // 3600 < 10:
+        fmt_str = "0" + fmt_str
+    return fmt_str
```

Comment:
Thoughts on breaking these lines out into a helper function that can be used both here and in `smartsim.settings.slurmSettings.SrunSettings._fmt_walltime`? That way if we have to update how to format the walltime for slurm we only need to touch one spot.

Reply:
hm hm `_fmt_walltime ` in `smartsim.settings.slurmSettings.SrunSettings` is never used, so Ill add it to `set_walltime` and break it out into a helper func, I was hesitant on that at first bc I wasnt sure about using helper funcs in outside folders but that makes more sense